File: Java/LeetCode/Calculate Digit Sum of a String.java
// https://leetcode.com/problems/calculate-digit-sum-of-a-string

class Solution {

    public String digitSum(String s, int k) {
        if (s.length() <= k) {
            return s;
        }

        StringBuilder newStr = new StringBuilder();
        for (int i = 0; i < s.length(); i += k) {
            String group = s.substring(i, Math.min(s.length(), i+k));

            int sum = 0;
            for (int j = 0; j < group.length(); ++j) {
                sum += Character.getNumericValue( group.charAt(j) );
            }

            newStr.append(String.valueOf(sum));
        }

        return digitSum(newStr.toString(), k);
    }

}

File: Kotlin/LeetCode/Calculate Digit Sum of a String.kt
// https://leetcode.com/problems/calculate-digit-sum-of-a-string

class Solution {

    fun digitSum(s: String, k: Int): String {
        if (s.length <= k) {
            return s
        }

        val newStr = StringBuilder()
        for (i in 0 until s.length step k) {
            val group = s.substring(i, Math.min(s.length, i+k))

            var sum = 0
            for (j in 0 until group.length) {
                sum += Character.getNumericValue(group[j])
            }

            newStr.append(sum.toString())
        }

        return digitSum(newStr.toString(), k)
    }

}
